testnet/stacks-node/src/nakamoto_node.rs

```diff
@@ -47,7 +47,7 @@ pub mod sign_coordinator;
 use self::peer::PeerThread;
 use self::relayer::{RelayerDirective, RelayerThread};
 
-pub const RELAYER_MAX_BUFFER: usize = 100;
+pub const RELAYER_MAX_BUFFER: usize = 1;
```

Comment:
This is kind of scary to me. Won't this cause the p2p stack to serialize with the relayer thread?

Reply:
I don't think so?  The changes to the p2p thread are to try and send the next `NetworkResult` to the relayer, and if the channel (with one item) is full, it instead holds onto it and merges subsequent `NetworkResult`'s into it via `NetworkResult::update()` until it can successfully send it via the channel.

A problem I noticed in testing prior to this patch is that without the above merge logic, the channel would very quickly fill up, and the p2p-thread's buffer would grow faster than the relayer could consume it.  This is because a lot of `NetworkResult`s contain useful data to the relayer, but that data is very sparse.  The `NetworkResult::update()` logic addresses this by squashing the queue down into a single `NetworkResult` that the p2p thread holds onto until the relayer is ready to receive it.